Dockerfile

```diff
@@ -0,0 +1,49 @@
+FROM php:8.3-apache
+
+# Update and install necessary packages
+RUN apt-get update && \
+    apt-get install -y libpng-dev libjpeg-dev libfreetype6-dev libopencv-dev && \
+    docker-php-ext-install gd && \
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Configure Apache
+RUN echo "<VirtualHost *:80>\n\
+    DocumentRoot /var/www/html\n\
+    <Directory /var/www/html>\n\
+    Options Indexes FollowSymLinks\n\
+    AllowOverride All\n\
+    Require all granted\n\
+    </Directory>\n\
+    # Redirect from root to Introduction.html\n\
+    RewriteEngine On\n\
+    RewriteCond %{REQUEST_URI} ^/$\n\
+    RewriteRule ^/$ /Introduction.html [R=301,L]\n\
+    </VirtualHost>" > /etc/apache2/sites-available/000-default.conf
+
+# Copy application files
+COPY ./build/ /var/www/html/
+COPY ../compile-all.sh /
```

Comment:
It'd be better to copy this file somewhere safer instead of in the root directly e.g. /tmp